std/stdio.d

```diff
@@ -1098,16 +1098,21 @@ Throws: `Exception` if the file is not opened.
         import std.conv : to, text;
         import std.exception : enforce, errnoEnforce;
 
+        // Some libc sanitize the whence input (e.g. glibc), but some don't,
+        // e.g. Microsoft runtime crashes on an invalid origin,
+        // and Musl additionally accept SEEK_DATA & SEEK_HOLE (Linux extension).
+        // To provide a consistent behavior cross platform, we use the glibc check
+        // See also https://issues.dlang.org/show_bug.cgi?id=19797
+        enforce(origin == SEEK_SET || origin == SEEK_CUR ||  origin == SEEK_END,
```

Comment:
Shouldn't this be an `assert` instead, as `origin` is not coming from user input, but from the developer that called this function?

Reply:
Well the case that sparked this was a programmer's error, so I think it's fair. @CyberShadow do you agree to change `enforce` to `assert` here ?

Reply:
We can only use `assert` in very particular circumstances in Phobos (namely template code which is *not* instantiated within Phobos) due to practical considerations (the compiler favoring the existing version in phobos.lib, which is not affected by the presence of `-release` on the user's command line), but otherwise I vacuously agree.